plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchRequestProcessor.java

```diff
@@ -358,11 +404,12 @@ private boolean validateQueryFieldInQueryString(
             for (Map<String, String> outputMap : processOutputMap) {
                 for (Map.Entry<String, String> entry : outputMap.entrySet()) {
                     String queryField = entry.getKey();
-                    Object pathData = jsonData.read(queryField);
-                    if (pathData == null) {
-                        throw new IllegalArgumentException(
-                            "cannot find field: " + queryField + " in query string: " + jsonData.jsonString()
-                        );
+                    // output writing to search extension can be new field
+                    if (!queryField.startsWith("ext.") && !queryField.startsWith("$.ext.")) {
+                        Object pathData = jsonData.read(queryField);
+                        if (pathData == null) {
+                            throw new IllegalArgumentException();
```

Comment:
please add more information so user can have more information on the cause and address the issue

Reply:
added error message